x/stakeibc/keeper/lsm_token_deposit_test.go

```diff
@@ -44,9 +54,134 @@ func (s *KeeperTestSuite) TestLSMTokenDepositRemove() {
+			"found amount %d did not match expected sum %d + %d = %d", actual.Amount,
+			existing.Amount, new.Amount, sdkmath.Int.Add(existing.Amount, new.Amount))
+	}
+
+	// verify that for previously non-existing deposits the amounts set
+	for i := 5; i < 10; i++ {
+		new := newDeposits[i]
+		actual, found := s.App.StakeibcKeeper.GetLSMTokenDeposit(s.Ctx, new.ChainId, new.Denom)
+		s.Require().True(found, "deposit not found in store %+v", new)
+		s.Require().Equal(actual.Amount, new.Amount,
+			"found amount %d did not match expected amount %d ", actual.Amount, new.Amount)
+	}
+}
+
+func (s *KeeperTestSuite) TestLSMTokenDepositStatusUpdate() {
+	statuses := []types.LSMDepositStatus{
+		types.TRANSFER_IN_PROGRESS,
+		types.TRANSFER_FAILED,
+		types.DETOKENIZATION_QUEUE,
+		types.DETOKENIZATION_IN_PROGRESS,
+		types.DETOKENIZATION_FAILED,
+	}
+	deposits := s.createSetNLSMTokenDeposit(5)
+	for i, status := range statuses {
+		s.App.StakeibcKeeper.UpdateLSMTokenDepositStatus(s.Ctx, deposits[i], status)
+	}
+
+	for i, deposit := range deposits {
+		actual, _ := s.App.StakeibcKeeper.GetLSMTokenDeposit(s.Ctx, deposit.ChainId, deposit.Denom)
+		s.Require().Equal(actual.Status, statuses[i], "status did not update for example %d", i)
+	}
+}
+
+func (s *KeeperTestSuite) TestLSMDepositsForHostZoneGet() {
+	// For HostZone with id i there will be i+1 deposits created, all denom unique
+	deposits := s.createNLSMTokenDeposit(15) // 15 = 1 + 2 + 3 + 4 + 5
+	idx := 0
+	for i := 0; i < 5; i++ {
+		for j := 0; j < i+1; j++ {
+			deposits[idx].ChainId = strconv.Itoa(i)
+			idx++
+		}
+	}
+	s.setGivenLSMTokenDeposit(deposits)
+
+	// Check there are i+1 deposits for chainid i, all deposits returned are from right chain
+	for i := 0; i < 5; i++ {
+		hostChainId := strconv.Itoa(i)
+		chainDeposits := s.App.StakeibcKeeper.GetLSMDepositsForHostZone(s.Ctx, hostChainId)
+		s.Require().Equal(len(chainDeposits), i+1, "Unexpected number of deposits found for chainId %d", i)
+		for _, deposit := range chainDeposits {
+			s.Require().Equal(deposit.ChainId, hostChainId, "Got a deposit from the wrong chain!")
+		}
+	}
+}
+
+func (s *KeeperTestSuite) TestLSMDepositsForHostZoneWithStatusGet() {
+	// Check that we get every deposit which matches hostzone and status
+	// Check that we only get deposits which match hostzone and status
+	// Need a predictable, different, non-zero number of deposits for each (zone,status) combo
+	// Deposit status is actually a 0 indexed int32 so make (hostzone+1)*(status+1) deposits
+	statuses := []types.LSMDepositStatus{
+		types.TRANSFER_IN_PROGRESS,
+		types.TRANSFER_FAILED,
+		types.DETOKENIZATION_QUEUE,
+		types.DETOKENIZATION_IN_PROGRESS,
+		types.DETOKENIZATION_FAILED,
+	}
+	deposits := s.createNLSMTokenDeposit(225) // total number across all combos
+	idx := 0
+	for hzid := 0; hzid < 5; hzid++ {
+		for sid := 0; sid < len(statuses); sid++ {
+			numCombo := (hzid + 1) * (sid + 1)
```

Comment:
I found this test a bit hard to follow, some suggestions that might clarify things a bit:
```suggestion
	// Check that we get every deposit which matches hostzone and status
	// Check that we only get deposits which match hostzone and status
	// Need a predictable, different, non-zero number of deposits for each (zone,status) combo
	numHostZones := 5
        statuses := []types.LSMDepositStatus{
		types.TRANSFER_IN_PROGRESS,
		types.TRANSFER_FAILED,
		types.DETOKENIZATION_QUEUE,
		types.DETOKENIZATION_IN_PROGRESS,
		types.DETOKENIZATION_FAILED,
	}
	
	// For each host zone + status combo, create a different number of deposits
	// that's determined by the product of (hostZone index + 1) * (status index + 1)
	// For instance:
	//    chain-0, status-0 => 1 deposit
	//    chain-0, status-1 => 1 * 2 = 2 deposit
	//    chain-1, status-1 => 2 * 2 = 4 deposits
	deposits := s.createNLSMTokenDeposit(225) // total number across all combos
	idx := 0
	for hzid := 0; hzid < numHostZones; hzid++ {
		for sid := 0; sid < len(statuses); sid++ {
			numCombo := (hzid + 1) * (sid + 1)
```

I was also trying to think if there was a way to dynamically calculate the 225 value based on the status' and host zones. For instance, the DEPOSIT_PENDING status was just added - it would be nice if this test did not have to be modified.
I couldn't think of how to do it using just an equation. Can you?

Reply:
yes, 15 came from the 5th triangular number T_n = 15 
and in general of course the nth triangular number T_n =  n*(n+1)/2 by the Gauss sum trick.
225 = 15^2 = T_5*T_5 and if instead we had 6 status and 5 hostzones then it would be T_6*T_5 = 21*15 = 315
Draw it as a table and in each cell is the number which is the product, but each row or column is just a multiple of the other direction's triangular number.  Visualizing it like this makes it easy to see why it is the multiplication.
I don't think this really helps make things clearer than a comment just saying this number is the total deposits needed.

Reply:
ah nice - thanks for the explainer - been awhile since I've done that type of math haha

And agreed, the formula is complex enough that I don't think you need to code it up. But maybe make a note to yourself to update this test once the commit with the DEPOSIT_PENDING status is merged in